src/replication/ng/adapter/adapter.go

```diff
@@ -67,6 +69,12 @@ func GetFactory(t model.RegistryType) (Factory, error) {
 	return factory, nil
 }
 
+// HasFactory checks whether there is given type adapter factory
```

Comment:
Why we need this method? I think we can just call `adapter.GetFactory(r.Type)` to get the factory, if it doesn't exist, an error will be returned

Reply:
We can use GetFactory, but does an error of it always means not exist, I think it's not explicit in semantics.